packages/aws-cdk-lib/core/lib/removal-policies.ts

```diff
@@ -0,0 +1,153 @@
+import { IConstruct } from 'constructs';
+import { Aspects, IAspect, AspectPriority } from './aspect';
+import { CfnResource } from './cfn-resource';
+import { RemovalPolicy } from './removal-policy';
+
+/**
+ * Properties for applying a removal policy
+ */
+export interface RemovalPolicyProps {
+  /**
+   * Apply the removal policy only to specific resource types.
+   * Can be a CloudFormation resource type string (e.g., 'AWS::S3::Bucket').
+   * @default - apply to all resources
+   */
+  readonly applyToResourceTypes?: string[];
+
+  /**
+   * Exclude specific resource types from the removal policy.
+   * Can be a CloudFormation resource type string (e.g., 'AWS::S3::Bucket').
+   * @default - no exclusions
+   */
+  readonly excludeResourceTypes?: string[];
+
+  /**
+   * If true, overwrite any user-specified removal policy that has been previously set.
+   * This means even if the user has already called `applyRemovalPolicy()` on the resource,
+   * this aspect will override it.
+   * @default false - do not overwrite user-specified policies
+   */
+  readonly overwrite?: boolean;
+
+  /**
+   * The priority to use when applying this aspect.
+   * If multiple aspects apply conflicting settings, the one with the higher priority wins.
```

Comment:
I'm trying to think of one example.

```suggestion
   * The priority to use when applying this aspect.
   *
   * The higher priority (lower numeric value) is applied first, and the lower priority
   * (higher numeric value) is applied later.
   * If multiple removal policies are applied with conflicting settings, the one with
   * the higher priority will be applied. However, if `overwrite` is true, the other policy
   * will overwrite it.
```

Reply:
see: https://github.com/aws/aws-cdk/pull/32283#discussion_r1897604240